File: js/lib/schemas/BooleanSchema.ts
import { BooleanFalseRule, BooleanTrueRule } from '../rules/booleanRules';
import { BaseSchema } from './BaseSchema';

export class BooleanSchema extends BaseSchema<boolean> {
  constructor() {
    super('number');
  }

  true() {
    return this.addRule(new BooleanTrueRule());
  }

  false() {
    return this.addRule(new BooleanFalseRule());
  }
}

File: net/lib/schemas/BooleanSchema.cs
using Albert.Lib.Rules;

namespace Albert.Lib.Schemas;

public class BooleanSchema : BaseSchema<bool>
{
    public override string Type => "string";

    public BooleanSchema True()
    {
        this.AddRule(new BooleanTrueRule());
        return this;
    }

    public BooleanSchema Max()
    {
        this.AddRule(new BooleanFalseRule());
        return this;
    }
}